packages/api/src/routes/medical/internal-cq.ts

```diff
@@ -192,23 +138,21 @@ router.put(
   requestLogger,
   asyncHandler(async (req: Request, res: Response) => {
     if (Config.isSandbox()) return res.sendStatus(httpStatus.NOT_IMPLEMENTED);
-    const cq = makeCarequalityManagementAPI();
-    if (!cq) throw new Error("Carequality API not initialized");
     const cxId = getUUIDFrom("query", req, "cxId").orFail();
     const oid = getFrom("params").orFail("oid", req);
-    await verifyCxProviderAccess(cxId);
-
     const org = await getOrganizationByOidOrFail({ cxId, oid });
     if (!org.cqApproved) throw new NotFoundError("CQ not approved");
+    await verifyCxProviderAccess(org);
 
```

Comment:
Since this is for internal usage, I reckoned we could avoid a hit to the DB and first load the Org - potentially exposing a 404 to the user even though the provided `cxId` is not associated with the requested Org.